bundles/org.openhab.binding.ism8/src/main/java/org/openhab/binding/ism8/server/Server.java

```diff
@@ -0,0 +1,281 @@
+     *
+     */
+    public void sendData(byte[] data) throws IOException {
+        Socket clientSocket = this.client;
+        if (clientSocket != null && clientSocket.isConnected() && data.length > 0) {
+            OutputStream stream = clientSocket.getOutputStream();
+            stream.write(data);
+            logger.debug("Data sent: {}", this.printBytes(data));
+        }
+    }
+
+    private void stopServer() {
+        logger.info("Stop Ism8 server.");
+        try {
+            ServerSocket serverSock = this.serverSocket;
+            if (serverSock != null) {
+                serverSock.close();
+            }
+
+            Socket clientSocket = this.client;
+            if (clientSocket != null) {
+                if (clientSocket.isConnected()) {
+                    clientSocket.getInputStream().close();
+                }
+
+                clientSocket.close();
+                this.client = null;
+            }
+        } catch (Exception e) {
+            logger.warn("Error stopping Communication. {}", e.getMessage(), e);
+        }
+    }
+
+    private void handleCommunication() {
+        try {
+            logger.info("Waiting for connection in port {}.", this.getPort());
+            IDataPointChangeListener listener = this.changeListener;
+            if (listener != null) {
+                listener.connectionStatusChanged(ThingStatus.OFFLINE);
+            }
+            
+            ServerSocket serverSock =new java.net.ServerSocket(this.getPort());
+            this.serverSocket = serverSock;
+            Socket clientSocket = serverSock.accept();
+            this.client = clientSocket;
+            logger.info("Connection from Partner established {}", clientSocket.getRemoteSocketAddress());
+            if (listener != null) {
+                listener.connectionStatusChanged(ThingStatus.ONLINE);
+            }
+
+            this.startRetries = 0;
+            this.sendUpdateCommand();
+            while (!this.isInterrupted()) {
+                byte[] bytes = getBytesFromInputStream(clientSocket.getInputStream());
+                int amount = bytes.length;
+                ArrayList<byte[]> packages = this.getPackages(bytes, amount);
+
+                for (byte[] pack : packages) {
+                    logger.debug("Data received: {}", this.printBytes(pack));
+                    KnxNetFrame frame = KnxNetFrame.createKnxNetPackage(pack, pack.length);
+                    if (frame != null) {
+                        byte[] answer = frame.createFrameAnswer();
+                        if (answer.length > 0) {
+                            this.sendData(answer);
+                        }
+                        SetDatapointValueMessage[] messages = frame.getValueMessages();
+                        for (int j = 0; j < messages.length; j++) {
+                            logger.debug("Message received: {} {}", messages[j].getId(),
+                                    this.printBytes(messages[j].getData()));
```

Comment:
Syntactical sugar. Same for below.
```suggestion
                        for (SetDatapointValueMessage message : frame.getValueMessages()) {
                            logger.debug("Message received: {} {}", message.getId(),
                                    this.printBytes(message.getData()));
```

Reply:
Changed. Thanks